src/routers/features/cgn/merchants.ts

```diff
@@ -197,17 +200,19 @@ addHandler(
           };
 
           const discountOption = () => {
-            switch (discountType) {
+            switch (foundMerchant.discountCodeType) {
               case "static":
                 return {
                   staticCode: faker.datatype
                     .string()
                     .toString() as NonEmptyString
                 };
-              case "landing":
+              case "landingpage":
                 return {
-                  landingPageReferrer: faker.datatype.string(6),
-                  landingPageUrl: "http://localhost:3000/merchant_landing"
+                  landingPageReferrer: faker.datatype.string(
+                    6
+                  ) as Discount["landingPageReferrer"],
+                  landingPageUrl: "http://localhost:3000/merchant_landing" as Discount["landingPageUrl"]
```

Comment:
```suggestion
                  landingPageUrl: `http://${interfaces.name}:${serverPort}/merchant_landing` as Discount["landingPageUrl"]
```
Not related with these changes but I guess it is an improvement to avoid hardcoded addresses